### DISCUSSION TITLE
Difference between current of the motor on teach Pendant screen and /joint_states values (Effort)?

### DISCUSSION THREAD

**Discussion by MohamedOmar2014** [ros-industrial/universal_robot#661] [Q&A]
I'm working on a real UR10 robot and I concern measuring the current of motor for further analysis of the motor torqe because i cant get it directly. I noticed that their is difference between current values on teach pendant screen and the values from the joint_states topic for a given configuration of the robot as shown in the following photos.
The current values (joint_states): -4.1294   -9.2880   -0.1740    -0.4531    0.2624    -0.0493 
The current values (Screen):         0.2           9.3          3.9            0.5          0.2           0
It seems that in the joint_states, the current value of joint 1 is replaced with current value of joint 3 because the current value of joint 3 has to be bigger based on the robot configuration.

Is that right or their is something missing?
thanks in advance

![Weixin Image_20230828211314](https://github.com/ros-industrial/universal_robot/assets/120853123/47f3f167-05c2-4613-af25-928b4da19f0c)
![Weixin Image_20230828211320](https://github.com/ros-industrial/universal_robot/assets/120853123/254fddab-dd75-4b01-9bac-a652f6da6a9d)
![Weixin Image_20230828211302](https://github.com/ros-industrial/universal_robot/assets/120853123/1f9c8ff0-86dd-404f-8bdf-37ba8c7fbeaf)

**Comment by RobertWilbrandt** [comment]
Did you check the order of the joints in the `name` field of the [JointState message](http://docs.ros.org/en/api/sensor_msgs/html/msg/JointState.html)? This should be consistent with the order of the positions/velocities/efforts and you cannot rely on the order of those fields not changing.
